aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -38,6 +40,8 @@
 
 public class AwsClientFactories {
 
+  public static final String APACHE_HTTP_TYPE = "apache";
```

Comment:
this should be in `AwsProperties`, with variables `CLIENT_HTTP_TYPE_APACHE`, `CLIENT_HTTP_TYPE_URL`, `CLIENT_HTTP_TYPE_DEFAULT`